### PR TITLE
feat(engine): Cold Start - Primary Parameters

### LINKED ISSUE #2351: [BUG] Fuel flow rises instantly from 0 to Idle flow
**Mod Version**

master

**Describe the bug**

fuel flow instantly rises from 0 to idle fuel flow (also wrong value - real is **~300-320kg/h per ENG**) 

**To Reproduce**

1.make a engine start - look @ MFD fuel flow indication
2.
3.

**Expected behavior**

fuel flow should rise, connected to N2, and not from 0 instantly to idle fuel flow, while N2 is rising.

**Actual behavior**

fuel flow rises 0 to idle fuel flow instandly

**References**

**Additional context**

Was this working before/when did the issue start occurring?
no
Is this a problem in the vanilla unmodded game?

Discord username (if different from GitHub):

### LINKED ISSUE #4621: During start of Engine 1, Engine 2 consumes negative fuel
**Aircraft Version**
Development 8cc7d5f

**Describe the bug**
When starting engine, Engine 1 does not consume any fuel and Engine 2 has negative fuel flow

**To Reproduce**
1. Start APU, APU bleed etc.
2. Turn on IGN mode
3. Switch Engine 1 master switch

**Expected behavior**
Engine 1 starts consuming fuel

**Actual behavior**
Engine 2 consumes negative fuel (start producing fuel?)

**Additional context**
Was this working before/when did the issue start occurring?

It worked in version before  c43d026

Is this a problem in the vanilla unmodded game?

Yes

### LINKED ISSUE #5001: Fuel Flow no longer ramping up during engine start sequence + EGT bump
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-05-28T15:17:02+00:00",
  "ref": "merge",
  "sha": "e6e2c8e772f6f835c8b9b251236e1df1b9b464f2",
  "actor": "tracernz",
  "event_name": "manual"
}
```

### Describe the bug

The fix provided by PR #4658 is no longer working during engine start.
During the start sequence, fuel flow is no longer ramping up. The fuel flow is only displayed post start-up sequence (as AVAIL disappears) alongside with a huge EGT bump (412° to 530° in a split second). 

### Expected behavior

Expectations are: 
- Fuel ramping up during start sequence (as per PR #4658, around 40%N1 for start point due to FADEC model)
- EGT slowly rising as fuel is being introduced by FADEC.  

### Steps to reproduce

1. Perform an ENG 1 or ENG 2 start with APU. 
2. Check behavior of fuel flow on ECAM display
3. Check EGT bump

### References (optional)

PR #4658 is the main reference

### Additional info (optional)

Discovered post update to 1.16. It was working properly before. 

### Discord Username (optional)

Remi#8871

### PR DESCRIPTION
**Pull request by Taz5150**
Fixes #5001
Fixes #4621 
Fixes #2351

## Summary of Changes

This PR features a realistic cold engine start-up process for the primary parameters (N1, N2, EGT and FF). Engine spools as a function of pressure and temperature until Idle condition is reached.

No additional visuals have been implemented/ changed, except for the small delay in the start-valve opening (SD Engine page).

Sound XML file updated and adapted to new LVars (courtesy of boris#8008).

New Engine LVars included in the FDR code (courtesy of AGuther [Z+2]#0001).

## Screenshots (if necessary)

Before engine start, all parameters are stable at 0 except EGT which is set at ambient temperature.

EWD-1

Expected behavior of N2, EGT and FF as a function of time during start-up:

image

## References

- Several YT videos of real Leap start-ups
- Discord user FlyingCarpet7#7502

## Additional context

Discord username for Engine Model (if different from GitHub): TazX [Z+2]#0405
Discord username for Sound Model (if different from GitHub): boris#8008

## Testing instructions

For testing, check that the start-up procedure is as expected in the figures above for different set of environmental conditions:

- High and low altitude airports
- High and low ambient temperatures

Other than the process itself:
- check that EGT initial temperature starts at ambient temperature
- check that the transition from start-up to Idle is smooth (since two separate functions are being used)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 1]
```diff
@@ -2003,19 +2005,15 @@ var A320_Neo_UpperECAM;
...
+            const value = SimVar.GetSimVarValue("L:A32NX_ENGINE_EGT:" + engineId, "celsius");
             return value;
         }
         getN1GaugeValue() {
             const engineId = (this.index + 1);
-            const value = SimVar.GetSimVarValue("ENG N1 RPM:" + engineId, "percent");
+            const value = SimVar.GetSimVarValue("L:A32NX_ENGINE_N1:" + engineId, "percent");
+            if (value < 0.0) {
+                value = 0;
+            }
```
Comment (MisterChocker) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2016]: This doesn't work, you'll need to use a let instead.

**Review by MisterChocker — CHANGES_REQUESTED** [review]

File: flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/engines.cfg [hunk 2]
```diff
@@ -107,4 +109,4 @@ thrust_scalar = 1
 DisableFuelValveControls = 1 ; Disable fuel valve controls
 DisableMixtureControls = 1 ; Disable mixture controls
 DisableParkingBrakeControls = 0 ; Disable parking brake controls
-DisablePropellerControls = 0 ; Disable propeller controls
+DisablePropellerControls = 0 ; Disable propeller controls
```
Comment (aguther) [inline flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/engines.cfg:112]: avoid unnecessary changes

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 3]
```diff
@@ -2198,15 +2196,8 @@ var A320_Neo_UpperECAM;
...
-            const imbalance = SimVar.GetSimVarValue("L:A32NX_ENGINE_IMBALANCE", "number");
-            let n2Imbalance = 0;
-            if (parseInt(imbalance.toString().substr(0, 1)) == _engine) {
-                n2Imbalance = parseFloat(imbalance.toString().substr(5, 2)) / 100;
-            } else {
-                n2Imbalance = parseFloat(0);
-            }
-            const name = "ENG N2 RPM:" + _engine;
-            let percent = SimVar.GetSimVarValue(name, "percent") - n2Imbalance;
+            const name = "L:A32NX_ENGINE_N2:" + _engine;
```
Comment (aguther) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2199]: creation of name can be done inline like for N1

**Review by aguther — COMMENTED** [review]
Please document any new local variables. Should they be added to FDR as well?

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 4]
```diff
@@ -2003,19 +2005,15 @@ var A320_Neo_UpperECAM;
...
             return value;
         }
         getN1GaugeValue() {
             const engineId = (this.index + 1);
-            const value = SimVar.GetSimVarValue("ENG N1 RPM:" + engineId, "percent");
+            let value = SimVar.GetSimVarValue("L:A32NX_ENGINE_N1:" + engineId, "percent");
+            if (value < 0.0) {
+                value = 0;
+            }
             return value;
```
Comment (lukecologne) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2017]:
```suggestion
            return Math.max(SimVar.GetSimVarValue("L:A32NX_ENGINE_N1:" + engineId, "percent"), 0);
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 5]
```diff
@@ -2003,19 +2005,15 @@ var A320_Neo_UpperECAM;
...
-            const imbalance = SimVar.GetSimVarValue("L:A32NX_ENGINE_IMBALANCE", "number");
-            let egtImbalance = 0;
-            if (parseInt(imbalance.toString().substr(0, 1)) == engineId) {
-                egtImbalance = parseInt(imbalance.toString().substr(1, 2));
-            } else {
-                egtImbalance = parseInt(0);
-            }
-            const value = SimVar.GetSimVarValue("L:A32NX_ENGINE_EGT:" + engineId, "celsius") - egtImbalance;
+            const value = SimVar.GetSimVarValue("L:A32NX_ENGINE_EGT:" + engineId, "celsius");
             return value;
```
Comment (lukecologne) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2009]:
```suggestion
            return SimVar.GetSimVarValue("L:A32NX_ENGINE_EGT:" + engineId, "celsius");
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
+// TazX - 06/08/2021: Engine imbalance has been hardcoded in the Wasm. All references have been eliminated.
```
Comment (lukecologne) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:1]: You don't need to leave comments on what you did, `git blame` can do that

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 7]
```diff
@@ -2198,15 +2196,8 @@ var A320_Neo_UpperECAM;
...
-            let n2Imbalance = 0;
-            if (parseInt(imbalance.toString().substr(0, 1)) == _engine) {
-                n2Imbalance = parseFloat(imbalance.toString().substr(5, 2)) / 100;
-            } else {
-                n2Imbalance = parseFloat(0);
-            }
-            const name = "ENG N2 RPM:" + _engine;
-            let percent = SimVar.GetSimVarValue(name, "percent") - n2Imbalance;
+            const name = "L:A32NX_ENGINE_N2:" + _engine;
+            let percent = SimVar.GetSimVarValue(name, "percent");
```
Comment (lukecologne) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2200]: Just directly return, and use `Math.max()` like I commented above

File: flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/engines.cfg (hunk 2 above)
Comment (Taz5150) [reply flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/engines.cfg:112]: Not sure what happened there, honestly.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js (hunk 1 above)
Comment (Taz5150) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js]: Done! Thanks.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js (hunk 3 above)
Comment (Taz5150) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js]: Change committed.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js (hunk 6 above)
Comment (Taz5150) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js]: Deleted comments on both Upper and Lower ECAM js files.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js (hunk 7 above)
Comment (Taz5150) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js]: Done!

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 8]
```diff
@@ -2198,19 +2189,7 @@ var A320_Neo_UpperECAM;
...
-            } else {
-                n2Imbalance = parseFloat(0);
-            }
-            const name = "ENG N2 RPM:" + _engine;
-            let percent = SimVar.GetSimVarValue(name, "percent") - n2Imbalance;
-            if (percent < 0.0) {
-                percent = 0;
-            }
-            return percent;
+			return Math.max(SimVar.GetSimVarValue("L:A32NX_ENGINE_N2:" + engineId, "percent"), 0);
```
Comment (lukecologne) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2192]: Looks like my code suggestion had an indentation error, sorry for that

Comment (Taz5150) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js]: No problem. Just solved it!! :)

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 9]
```diff
@@ -2198,19 +2189,7 @@ var A320_Neo_UpperECAM;
...
-            } else {
-                n2Imbalance = parseFloat(0);
-            }
-            const name = "ENG N2 RPM:" + _engine;
-            let percent = SimVar.GetSimVarValue(name, "percent") - n2Imbalance;
-            if (percent < 0.0) {
-                percent = 0;
-            }
-            return percent;
+            return Math.max(SimVar.GetSimVarValue("L:A32NX_ENGINE_N2:" + engineId, "percent"), 0);
```
Comment (aguther) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2192]:
```suggestion
            const engineId = this.index + 1;
            return Math.max(SimVar.GetSimVarValue("L:A32NX_ENGINE_N2:" + engineId, "percent"), 0);
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 10]
```diff
@@ -2198,19 +2189,8 @@ var A320_Neo_UpperECAM;
...
-                n2Imbalance = parseFloat(0);
-            }
-            const name = "ENG N2 RPM:" + _engine;
-            let percent = SimVar.GetSimVarValue(name, "percent") - n2Imbalance;
-            if (percent < 0.0) {
-                percent = 0;
-            }
-            return percent;
+            const engineId = this.index + 1;
+            return Math.max(SimVar.GetSimVarValue("L:A32NX_ENGINE_N2:" + engineId, "percent"), 0);
```
Comment (aguther) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2193]:
```suggestion
            return Math.max(SimVar.GetSimVarValue("L:A32NX_ENGINE_N2:" + _engine, "percent"), 0);
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 11]
```diff
@@ -2198,19 +2189,8 @@ var A320_Neo_UpperECAM;
...
-            } else {
-                n2Imbalance = parseFloat(0);
-            }
-            const name = "ENG N2 RPM:" + _engine;
-            let percent = SimVar.GetSimVarValue(name, "percent") - n2Imbalance;
-            if (percent < 0.0) {
-                percent = 0;
-            }
-            return percent;
+            const engineId = this.index + 1;
```
Comment (aguther) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:2192]:
```suggestion
```

File: src/fdr2csv/src/main.cpp [hunk 12]
```diff
@@ -14,7 +14,7 @@
 using namespace std;
 
 // IMPORTANT: this constant needs to increased with every interface change
-const uint64_t INTERFACE_VERSION = 6;
+const uint64_t INTERFACE_VERSION = 7;
```
Comment (aguther) [inline src/fdr2csv/src/main.cpp:17]:
```suggestion
const uint64_t INTERFACE_VERSION = 8;
```

File: src/fbw/src/FlightDataRecorder.h [hunk 13]
```diff
@@ -12,7 +12,7 @@
 class FlightDataRecorder {
  public:
   // IMPORTANT: this constant needs to increased with every interface change
-  const uint64_t INTERFACE_VERSION = 6;
+  const uint64_t INTERFACE_VERSION = 7;
```
Comment (aguther) [inline src/fbw/src/FlightDataRecorder.h:15]:
```suggestion
  const uint64_t INTERFACE_VERSION = 8;
```

**Review by aguther — APPROVED** [review]

**Comment by Samueleonard** [comment]
Quality Assurance Tester Report

Discord: 1samueleonard1
Object of testing: (#5081)
Tier of Testing: (1)
Date: (12/06/2021)

Testing Process:

. loaded up and set up as normal
. started engines and observed NORMAL CONDITIONS - looks fine
. shut down engines and observed - looks fine however EGT doesnt drop below about 400 for a few minutes, and then drops immediately to OAT

. changed weather to extreme hot (+40)
. started + shutdown - startup ok,  EGT issue

. changed weather to extreme cold (-40)
. started + shutdown - engines started but never got an AVAIL light on either engine, same issue with EGT above

. moved to high altitude airport (la paz)
. started + shutdown normal conditions - NOT TESTED YET
. repeated with extreme high + low temps - NOT TESTED YET

Negatives: as above + oil psi goes past the max indicator line, but continues to rise (normal?)
Testing Results: (Not Passed)

Conclusions: needs work, as written above
Media:

**Comment by frankkopp** [comment]
Quality Assurance Tester Report

Discord: Cdr_Maverick#6475
Object of testing: #5081
Tier of Testing: 1
Date: 12.6.2021

Testing Process:
For testing, check that the start-up procedure is as expected in the figures above for different set of environmental conditions:
High and low altitude airports
High and low ambient temperatures
Other than the process itself:
check that EGT initial temperature starts at ambient temperature
check that the transition from start-up to Idle is smooth (since two separate functions are being used)

1. Normal conditions +15° ==> OK
2. Cold -40° ==> OK (but ECAM does never show AVAIL for engines - N1 stays below 18% (see video)
3. Hot +45° ==> OK 
4. High + Cold:  VNKT 4358ft  -40° ==> OK ( but no AVAIL msg)
5. Low + Hot:   LLMZ -358m (-1250ft) ==> OK

Negatives: 
-Startup OK
- Shutdown EGT stays at one value for about 1min then jumps to OAT
- When very cold ECAM never shows AVAIL for engines. 

Testing Results: ??? - is the cold+NON-AVAIL an issue or realistic?

Conclusions: 
Subjective impression is that the startup works as intended and according to the PR. 
The EGT after shutdown issue found by 1samueleonard1 is confirmed although probably not part of this PR.

Media:
Normal https://youtu.be/uG-vlb8HbnA
Cold https://youtu.be/I2NcpzHZjQw
Hot https://youtu.be/2d5UNxNONkU
High Cold https://youtu.be/8LoEYJNK2fM
Low Hot https://youtu.be/zgrJk90X3FQ

**Comment by Taz5150** [comment]
> Quality Assurance Tester Report
> 
> Discord: 1samueleonard1
> Object of testing: (#5081)
> Tier of Testing: (1)
> Date: (12/06/2021)
> 
> Testing Process:
> 
> . loaded up and set up as normal
> . started engines and observed NORMAL CONDITIONS - looks fine
> . shut down engines and observed - looks fine however EGT doesnt drop below about 400 for a few minutes, and then drops immediately to OAT
> 
> . changed weather to extreme hot (+40)
> . started + shutdown - startup ok, EGT issue
> 
> . changed weather to extreme cold (-40)
> . started + shutdown - engines started but never got an AVAIL light on either engine, same issue with EGT above
> 
> . moved to high altitude airport (la paz - 13000ft)
> . started + shutdown normal conditions - startup ok, shutdown EGT issue
> . repeated with extreme high + low temps - NOT TESTED YET
> 
> Negatives: as above + oil psi goes past the max indicator line, but continues to rise (normal?)
> Testing Results: (Not Passed)
> 
> Conclusions: needs work, as written above - cannot verify values are correct, would need real pilot
> Media:

Thank you for the test and feedback!

Only engine start and primary parameters have been modeled so far. This means that:

- Engine shutdown parameters can be (will be) all over the place.
- Secondary engine parameters (everything within the Engine page in the SD) are not modeled and thus maintain original/ vanilla aircraft values.

Regarding AVAIL on the EWD, it is still programmed to appear at N1 = 18%. Although visuals are still not been taken care of, I think it would be beneficial to work on the AVAIL issue and include it in this PR. Will do so and probably have it ready today (06/13).

Thank you again for the effort!

**Comment by Taz5150** [comment]
> Quality Assurance Tester Report
> 
> Discord: Cdr_Maverick#6475
> Object of testing: #5081
> Tier of Testing: 1
> Date: 12.6.2021
> 
> Testing Process:
> For testing, check that the start-up procedure is as expected in the figures above for different set of environmental conditions:
> High and low altitude airports
> High and low ambient temperatures
> Other than the process itself:
> check that EGT initial temperature starts at ambient temperature
> check that the transition from start-up to Idle is smooth (since two separate functions are being used)
> 
> 1. Normal conditions +15° ==> OK
> 2. Cold -40° ==> OK (but ECAM does never show AVAIL for engines - N1 stays below 18% (see video)
> 3. Hot +45° ==> OK
> 4. High + Cold:  VNKT 4358ft  -40° ==> OK ( but no AVAIL msg)
> 5. Low + Hot:   LLMZ -358m (-1250ft) ==> OK
> 
> Negatives:
> -Startup OK
> 
> * Shutdown EGT stays at one value for about 1min then jumps to OAT
> * When very cold ECAM never shows AVAIL for engines.
> 
> Testing Results: ??? - is the cold+NON-AVAIL an issue or realistic?
> 
> Conclusions:
> Subjective impression is that the startup works as intended and according to the PR.
> The EGT after shutdown issue found by 1samueleonard1 is confirmed although probably not part of this PR.
> 
> Media:
> Normal https://youtu.be/uG-vlb8HbnA
> Cold https://youtu.be/I2NcpzHZjQw
> Hot https://youtu.be/2d5UNxNONkU
> High Cold https://youtu.be/8LoEYJNK2fM
> Low Hot https://youtu.be/zgrJk90X3FQ

Thank you for the test!

Same comments apply as before. Shutdown needs to be modeled and AVAIL is considered important enough to have it worked on and included in this PR.

Thank you!

**Comment by Taz5150** [comment]
UPDATE: AVAIL is now correctly modeled.

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #5081
Tier of Testing: 1
Date: 13/06/2021

Testing Process: 
- Spawn C&D at EDDP
- TAT 18°C
- Engine start up > OK
- Engine shut down
- TAT -60°C
- Engine start up > Freeze at N1 = 3.1% 
- Respawn
- TAT -60°C
- Engine start up > Freeze at N1 = 3.1% 
- TAT -42°C
- Engine start resumes
- TAT 54°C
- Engine start up > OK

Negatives: 
- N2 digit background change on start up (0% to about 58%) to from black to dark grey (saw that before, so not part of this PR)
- On TAT of -60°C start up get stuck on about N1 3.1%

Testing Results: Passed, as TAT -60°C is very unrealistic i think
Conclusions: Works
Media: -
Hope this helps :-)

**Comment by frankkopp** [comment]
Quality Assurance Tester Report

Discord: Cdr_Maverick#6475
Object of testing: #5081
Tier of Testing: 1
Date: 14.6.2021

Testing Process:

Only re-doing the cold test from above:
Cold -40° ==> OK and AVAIL is shown!

Negatives: none
Testing Results: Passed
Conclusions: LGTM

**Comment by Cryptic56** [comment]
Quality Assurance Tester/Trainee Report

Discord: Cryptic#4731
Object of testing: #5081 
Tier of Testing: 1
Date: 14/06/2021

Testing Process: Spawned CD at UEEE
Temperature -40
started APU and then engines
Observed Startup (Good)
Restart
Temperature +50 
started APU and then engines
Observed startup (Good)
Observed shutdown 


Negatives: On shutdown, the EGT stopped decreasing at about 350 degrees then a minute later dropped to OAT
When changing the temperature the engines EGT drops immediately to the OAT rather than slowly decreasing like the APU EGT
 
Testing Results: Passed

Conclusions: A few issues outlined above but other than that, worked great. Big improvement. Great work. 
(Sorry for the brief report, been a little busy this weekend)
